extension/lib/bigcommerce/RequestRepository.js

```diff
@@ -0,0 +1,92 @@
+const BigCommerceLogger = require('./Logger')
+
+class BigCommerceRequestRepository {
+  /**
+   * @param {BigCommerce} client Api V3 client
+   * @param {Logger} logger context.log
+   */
+  constructor (client, logger) {
+    this.client = client
+    this.logger = logger
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  get (path) {
+    return this.request('get', path)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  post (path, data, obfuscatedData = null) {
+    return this.request('post', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  put (path, data, obfuscatedData = null) {
+    return this.request('put', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  del (path) {
+    return this.request('delete', path)
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} path
+   * @param {Object} [data]
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  async request (type, path, data = null, obfuscatedData = null) {
```

Comment:
in this case you do no want to set null to these. It will be easier to do logging.  `undefined` is better because it drops the property completely, e.g. 
`JSON.stringify({a: null})`
results in `{"a":null}`

`JSON.stringify({a: undefined})`
results in `{}`

Reply:
if data is null, it will not be logged at all?!
that's part of the getRequestData function (which might need a better name)

Reply:
what I meant it would be cleaner if you implemented undefined because you can remove that if(data === null) { return { type, path } }` completely if data is 'undefined'. The reason for that is that your final `return { type, path, data }` will print `{type, path}` if data is undefined. Less code and complication in those terms.

Reply:
tested it. 
return { type, path, data }
still returns data as undefined and does not drop it.

Reply:
hm, info came from a channel full of js devs :D